go/vt/vttablet/grpctabletconn/conn.go

```diff
@@ -221,11 +221,7 @@ func (conn *gRPCQueryClient) Begin(ctx context.Context, target *querypb.Target,
 	if err != nil {
 		return 0, nil, tabletconn.ErrorFromGRPC(err)
 	}
-	// For backwards compatibility, we don't require tablet alias to be present in the response
-	// TODO(deepthi): After 7.0 change this
-	//	return br.TransactionId, br.TabletAlias, nil
-	// also assert that br.TabletAlias == conn.tablet.Alias
```

Comment:
Was the scope of this TODO to also assert `br.TabletAlias == conn.tablet.Alias`.

Reply:
I thought that was a good idea at that time, but we don't in general put assertions into production code.
We _could_ put an if condition that returns an error but it seems like overkill.

Reply:
@systay @harshit-gangal any opinions on this?

Reply:
I think this check is not necessary it is not giving any value here. as the conn here is the actual connection to the server it is unlikely it will differ.